### ISSUE TITLE
HiOp undefined reference to hiopMatrixSparseCSRCUDA

### ISSUE THREAD

**Issue by pghysels** [xsdk-project/xsdk-issues#191]
```
     364    /usr/bin/ld: /usr/bin/ld: ../src/libhiop.a(hiopLinAlgFactory.cpp.o): in function `hiop::LinearAlgebraFactory::create_matrix_sparse_csr(std:
            :__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
  >> 365    /tmp/pghysels/spack-stage/spack-stage-hiop-0.7.0-hocc47lrdmrvvsl5mgymneqbflymhnh7/spack-src/src/LinAlg/hiopLinAlgFactory.cpp:189: undefined
             reference to `hiop::hiopMatrixSparseCSRCUDA::hiopMatrixSparseCSRCUDA()'
     366    /usr/bin/ld: ../src/libhiop.a(hiopLinAlgFactory.cpp.o): in function `hiop::LinearAlgebraFactory::create_matrix_sparse_csr(std::__cxx11::bas
            ic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, int)':
```
I see this for HiOp 0.7.0 through spack, but also using manual git/cmake for hiop@develop.

However, with the manual CMake run, I see this:
```
-- Cannot find COINHSL, STRUMPACK, PARDISO nor GINKGO for sparse linear algebra, and the option HIOP_SPARSE will be disabled
```

In spack, even when I enable +ginkgo, HIOP_SPARSE is OFF, and I get the same error.

Do you think more changes are required in the hiop spack package for hiop v0.7.0?

[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/9804059/spack-build-out.txt)

**Comment by cnpetra** [comment]
for the build through spack, I see `-DHIOP_SPARSE:BOOL=OFF`, do you want it like this? The undefined reference apparently creeps in because of a flawed conditional. We will fix it shortly.

**Comment by cnpetra** [comment]
also, could you please send us the command and log for the manual CMake run?

**Comment by pghysels** [comment]
It looks like HIOP_SPARSE is OFF in spack by default because it relies on coinhsl which cannot be downloaded through spack?

The log for the cmake run is attached.
[hiop.log](https://github.com/xsdk-project/xsdk-issues/files/9804878/hiop.log)

**Comment by cnpetra** [comment]
> It looks like HIOP_SPARSE is OFF in spack by default because it relies on coinhsl which cannot be downloaded through spack?
> 
> The log for the cmake run is attached. [hiop.log](https://github.com/xsdk-project/xsdk-issues/files/9804878/hiop.log)

thanks. same underlying issue. 

I suggest we fix this and provide a new tag/release 0.7.1 for HiOp (with a couple of other build system fixes), would that work on your end?
